streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala

```diff
@@ -128,4 +128,37 @@ object ZStream extends Serializable {
         }
       } yield Control(pull, Command.noop)
     }
+
+  /**
+   * Creates a stream from an iterable collection of values
+   *
+   * @tparam A the iterable value type
+   * @return a stream which emits values from iterable
+   */
+  def fromIterable[A](as: => Iterable[A]): ZStream[Any, Nothing, Any, Unit, A] =
+    ZStream {
+      Managed.effectTotal {
+        Control(
+          IO.succeedNow(as.iterator).flatMap(iter => if (iter.hasNext) IO.succeed(iter.next()) else Pull.endUnit),
+          Command.noop
+        )
+      }
+    }
+
+  /**
+   * Creates a stream from an iterator
+   *
+   * @tparam A the value type of elements produced by iterator
+   * @return a stream which emits values produced by iterator
+   */
+  def fromIterator[A](iterator: => Iterator[A]): ZStream[Any, Nothing, Any, Unit, A] =
+    ZStream {
+      Managed.effectTotal {
+        Control(
+          IO.succeedNow(iterator).flatMap(iter => if (iter.hasNext) IO.succeed(iter.next()) else Pull.endUnit),
```

Comment:
Pulling from iterators is effectful, so this should be rewritten as:
```
IO.effectSuspendTotal {
  if (iter.hasNext) UIO(iter.next())
  else Pull.endUnit
}
```

Reply:
Shouldn't the `IO.succeedNow(iterator)` be better placed outside of `Control`?

Reply:
It's inside `Managed.effectTotal`, so I think we could just do...

```
Managed.effectTotal {
  val it = iterator
  Control(
    IO.effectSuspendTotal {
      // ...
    },
    Command.noop
}
```

Requires testing though.

Reply:
Changed argument type of method to `ZIO[R, E, Iterator[A]]` and implemented it in terms of `fromEffect` and `flatMap`